README.md

```diff
@@ -3,13 +3,22 @@
 [![crates.io](https://img.shields.io/crates/v/rustc-hash.svg)](https://crates.io/crates/rustc-hash)
 [![Documentation](https://docs.rs/rustc-hash/badge.svg)](https://docs.rs/rustc-hash)
 
-A speedy, non-cryptographic hashing algorithm used by `rustc` and Firefox.
+A speedy, non-cryptographic hashing algorithm used by `rustc`.
 The [hash map in `std`](https://doc.rust-lang.org/std/collections/struct.HashMap.html) uses SipHash by default, which provides resistance against DOS attacks.
-These attacks aren't as much of a concern in the compiler so we prefer to use the quicker, non-cryptographic Fx algorithm.
-
-The Fx algorithm is a unique one used by Firefox. It is fast because it can hash eight bytes at a time.
-Within `rustc`, it consistently outperforms every other tested algorithm (such as FNV).
-The collision rate is similar or slightly worse than other low-quality hash functions.
+These attacks aren't as much of a concern in the compiler so we prefer to use a
+quicker, non-cryptographic hash algorithm.
+
+The original hash algorithm provided by this crate was one taken from Firefox,
+hence the hasher it provides is called FxHasher. This name is kept for backwards
+compatibility, but the underlying hash has since been replaced. The current
+design for the hasher is a polynomial hash finished with a single bit rotation,
+together with a wyhash-inspired compression function for strings/slices, both
+designed by Orson Peters.
+
+Within `rustc`, it consistently outperforms every other tested algorithm or
+variation, despite its simplicity. Spending more CPU cycles on a higher quality
+hash does not reduce hash collisions enough to be worth the cost on real-world
+benchmarks.
```

Comment:
I found this paragraph confusing because the "it" used to apply to the old FxHash but now it applies to the new FxHash. So really there are two known hashing algorithms that give similar performance. Perhaps the paragraph should say:

"For `rustc` we have tried many different hashing algorithms. Hashing speed is critical, especially for single integers. Spending more CPU cycles on a higher quality
hash does not reduce hash collisions enough to make the compiler faster on real-world
benchmarks."